lib/cache/Cache.js

```diff
@@ -44,14 +45,33 @@ class Cache {
         return loc
     }
 
+    async write(filepath, data) {
+        const loc = this.getCacheLocation(filepath)
+        await mkdirp(path.dirname(loc))
```

Comment:
I didn't think you needed `path.dirname` here because I thought `mkdirp` did this automatically if you passed it a file with an extension rather than a directory.  However that's incorrect, so this line is right.  The added `mkdirp outloc`, however, might not be since it could be pointing at a file or directory, so that one (line `32`) should be `mkdirp(path.dirname(outLoc))` as well I believe...